src/main/java/securbank/controller/ManagerController.java

```diff
@@ -186,6 +199,70 @@ public String getUserDetails(Model model, @PathVariable UUID id) {
         return "manager/userdetail";
     }
 	
+	@GetMapping("/manager/transactions")
+    public String getTransactions(Model model) {
+
+		List<Transaction> transactions = transactionService.getTransactionsByStatus("Pending");
+		if (transactions == null) {
+			return "redirect:/error?code=500";
+		}
+		model.addAttribute("transactions", transactions);
+		logger.info("GET request:  All pending transactions");
+		
+        return "manager/pendingtransactions";
+    }
+	
+	@GetMapping("/manager/transaction/{id}")
+    public String getTransactionRequest(Model model, @PathVariable() UUID id) {
+		Transaction transaction = transactionService.getTransactionById(id);
+		
+		if (transaction == null) {
+			return "redirect:/error?code=404&path=request-invalid";
+		}
+
+		// checks if manager is authorized for the request to approve
+		if (!transaction.getAccount().getUser().getType().equals
+				("external")) {
+			logger.warn("GET request: Manager unauthrorised request access");
+			return "redirect:/error?code=401&path=request-unauthorised";
+		}
+		model.addAttribute("transaction", transaction);
+		logger.info("GET request: Manager external transaction request by ID");
+		
+        return "manager/approvetransaction";
+    }
+	
+	@PostMapping("/manager/transaction/request/{id}")
+    public String approveRejectTransactions(@ModelAttribute Transaction trans, @PathVariable() UUID id, BindingResult bindingResult) {
+		
+		Transaction transaction = transactionService.getTransactionById(id);
+		if (transaction == null) {
+			return "redirect:/error?code=404&path=request-invalid";
+		}
+		
+		// checks if manager is authorized for the request to approve
+		if (!transaction.getAccount().getUser().getType().equalsIgnoreCase
+				("external")) {
+			logger.warn("GET request: Manager unauthrorised request access");
+					
+			return "redirect:/error?code=401&path=request-unauthorised";
+		}
+		
+		if("approved".equalsIgnoreCase(trans.getApprovalStatus())){
+			if(transactionService.isTransactionValid(transaction)==false){
```

Comment:
@Nikh13 Have made your suggested changes and pushed them. Let me know if anything else needs to be changed before you merge this.